wallet/src/events.rs

```diff
@@ -4,12 +4,86 @@ use ic_cdk::export::Principal;
 use ic_cdk::{api, storage};
 use serde::Deserialize;
 use std::cmp::min;
+use std::collections::HashMap;
 
 #[derive(CandidType, Clone, Default, Deserialize)]
 pub struct EventBuffer {
     events: Vec<Event>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Default, CandidType, Deserialize)]
+pub struct ChildList(HashMap<Principal, Child>);
+
+impl ChildList {
+    pub fn push(&mut self, child: Principal, event: ChildEventKind) {
+        self.push_with_timestamp(child, event, api::time())
+    }
+    pub fn push_with_timestamp(&mut self, child: Principal, event: ChildEventKind, timestamp: u64) {
+        let events = &mut self
+            .0
+            .entry(child)
+            .or_insert_with(|| Child::new(child))
+            .events;
+        events.push(ChildEvent {
+            kind: event,
+            id: events.len() as u32,
+            timestamp,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Eq, CandidType, Deserialize)]
+pub struct Child {
+    pub info: ChildInfo,
+    pub events: Vec<ChildEvent>,
+}
```

Comment:
Naming things is hard...  In the context of the HashMap, this is a child at some level, but in the context of the overall domain, it's more about being a canister than being a child in a hashmap.
```suggestion
pub struct CanisterEntry {
    pub info: CanisterInfo,
    pub events: Vec<CanisterEvent>,
}
```